core/src/main/scala/org/http4s/EntityDecoder.scala

```diff
@@ -45,6 +45,56 @@ trait EntityDecoder[F[_], T] { self =>
       override def consumes: Set[MediaRange] = self.consumes
     }
 
+  def handleError(f: DecodeFailure => T)(implicit F: Functor[F]): EntityDecoder[F, T] =
```

Comment:
The following two methods seem strictly less powerful than `def transform[T2](t: Either[DecodeFailure, T] => Either[DecodeFailure, T2]` and `transformWith` in that you could implement it in terms of `transform` by using `identity` over the `Right` case.

Reply:
Excellent point. Reimplemented handleError and handleErrorWith using the transform methods.